main.go

```diff
@@ -98,40 +100,55 @@ func main() {
 	log.Printf(newRelease.GetHTMLURL())
 
 	if filelist := strings.TrimSpace(c.FilesToUpload); filelist != "" {
-		fmt.Println()
-		log.Infof("Uploading assets:")
-		files := strings.Split(filelist, "\n")
-		for i, filePath := range files {
-			if strings.TrimSpace(filePath) == "" {
-				continue
-			}
+		if err := uploadWithRetry(filelist, client, owner, repo, newRelease.GetID()); err != nil {
+			failf("Error during upload: %s", err)
+		}
+	}
+}
+
+func uploadWithRetry(filelist string, client *github.Client, owner string, repo string, id int64) error {
```

Comment:
NOTE: I would prefer to keep the parsing of the filelist in the main or an other function, to be more testable and create a function signature more clearly describing the functionality.

Reply:
Divided code to more parts, please check.